### DISCUSSION TITLE
Log retention

### DISCUSSION THREAD

**Discussion by GRRLjay** [cisagov/LME#528] [FAQ]
I was attempting to follow the guidance on adjusting retention. I don't have the lme_ilm_policy in my index lifecycle policies. Is that a problem with my instance or is the documentation out of date?

https://github.com/cisagov/LME/blob/main/docs/markdown/logging-guidance/retention.md

**Comment by aarz-snl** [comment]
That is older documentation for what the index used to be called. The steps are generally all the same. 

Check updated directions here:

https://github.com/cisagov/LME/blob/main/docs/markdown/maintenance/index-management.md

2 sections here -- one for creating a new policy to manage your wazuh logs -- and one for managing the existing policy that comes with elastic (logs that come in from elastic agents)


Ultimately, it will be the same concept. You're setting or editing a policy and then in that policy to move your index to the next stage after a certain amount of time or size. Completely up to you when you want to delete indexes based on these settings.

**Reply by GRRLjay** [reply]
Maybe I'm just not getting it, but the directions say to look for a lifecycle policy called "logs". However, when I search the lifecycle policies for "logs" I only get the following:

logs-endpoint.collection-diagnostic
logs-fleet-server.output_health_logs-default_policy
logs@lifecycle

Is there supposed to be something else that's missing?

I had some trouble with installation so I'm not confident that everything is as it should be.

**Reply by aarz-snl** [reply]
Did you follow step one and select: 

"Include managed system policies"

alot of managed polices / indices will be hidden on you hit a switch

Im also not entirely sure this get created once you install an agent on an endpoint or if it comes preloaded. Have you installed an agent on endpoint yet? I can check on that in my lab when I'm back in the office

**Reply by GRRLjay** [reply]
Yes, this is with that checked. There is just no "logs" only policy.

**Reply by GRRLjay** [reply]
Screenshot of Lifecycle Policies page:
![IndexLifecyclePolicies](https://github.com/user-attachments/assets/5ba4ee2a-c1bc-404f-bd29-dcfba8eff063)

**Reply by GRRLjay** [reply]
I found it. I had to click the Deprecated button. Is it right that it's deprecated when LME 2 was just released?

**Reply by aarz-snl** [reply]
Some things like this are managed by elastic - so if they deem it 'deprecated' they may have replaced it with something else. 

From what I can tell they replaced it to with the logs@lifecycle you see in your screenshot -- however, that has no linked indices to it. 

You certainly want to manage whichever ones actually has linked indices to it or your policy wont do anything. -- more research will be needed on this

**Reply by aarz-snl** [reply]
Here's more info on this -- doesn't appear youre the only confused person:

https://github.com/elastic/kibana/issues/201181

**Reply by aarz-snl** [reply]
My guess is they made this so it doesn't automatically convert in case users were upgrading from 8.11 to 8.12 it doesn't break their lifecycle policies they already setup with the other one. Keep in mind even if it says deprecated you can still use it it just means they wont make anymore updates to that lifecycle policy. You are self managing it at this point anyway though
